### PR TITLE
Add grain_size hyperparameter into EmpiricalCovariance and PCA algorithms

### PR DESCRIPTION
**Pull request by Vika-F**
- `cpu_grain_size` hyperparameter was added to `EmpiricalCovariance` estimator.
- `cpu_grain_size` and `cpu_macro_block` hyperparameters were added to `PCA` estimator.

- `gen_daal4py.py` was modified to report the problem with `clang-format` installation on the setup stage.

---

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [ ] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

not applicable

### CODE REVIEW

File: sklearnex/decomposition/tests/test_pca.py [hunk 1]
```diff
@@ -42,6 +44,14 @@ def test_sklearnex_import(dataframe, queue):
     ]
 
     pca = PCA(n_components=2, svd_solver="covariance_eigh")
+
+    if daal_check_version((2025, "P", 700)):
+        hparams = PCA.get_hyperparameters("fit")
+        if macro_block is not None:
```
Comment (david-cortes-intel) [inline sklearnex/decomposition/tests/test_pca.py]: Since all the tests are executed in the same process, this would leave the last parameters propagated to further tests after it is done.

Perhaps it could use some mechanism like in here:
https://github.com/uxlfoundation/scikit-learn-intelex/blob/cf2dbcbd8ed10bbbcd6b13116327b43d99c43fad/sklearnex/linear_model/tests/test_linear.py#L41

File: onedal/decomposition/pca.cpp [hunk 2]
```diff
@@ -221,6 +234,28 @@ void init_infer_ops(py::module_& m) {
...
+    auto cls = py::class_<train_hyperparams_t>(m, "train_hyperparameters")
+                   .def(py::init())
+                   .def("set_cpu_macro_block", &train_hyperparams_t::set_cpu_macro_block)
+                   .def("get_cpu_macro_block",
+                        [](const train_hyperparams_t& self) {
+                            return self.get_cpu_macro_block();
+                        })
+                   .def("set_cpu_grain_size", &train_hyperparams_t::set_cpu_grain_size)
+                   .def("get_cpu_grain_size", [](const train_hyperparams_t& self) {
+                       return self.get_cpu_grain_size();
```
Comment (icfaust) [inline onedal/decomposition/pca.cpp]:
```suggestion
                   .def("set_cpu_macro_block", &train_hyperparams_t::set_cpu_macro_block)
                   .def("get_cpu_macro_block", &train_hyperparams_t::get_cpu_macro_block)
                   .def("set_cpu_grain_size", &train_hyperparams_t::set_cpu_grain_size)
                   .def("get_cpu_grain_size", &train_hyperparams_t::.get_cpu_grain_size);
```

File: onedal/covariance/covariance.cpp [hunk 3]
```diff
@@ -173,8 +173,18 @@ void init_compute_hyperparameters(py::module_& m) {
...
+                        [](const compute_hyperparams_t& self) {
+                            return self.get_cpu_macro_block();
+#if defined(ONEDAL_VERSION) && ONEDAL_VERSION >= 20250700
+                        })
+                   .def("set_cpu_grain_size",
+                        [](compute_hyperparams_t& self, int64_t cpu_grain_size) {
+                            self.set_cpu_grain_size(cpu_grain_size);
+                        })
+                   .def("get_cpu_grain_size", [](const compute_hyperparams_t& self) {
+                       return self.get_cpu_grain_size();
```
Comment (icfaust) [inline onedal/covariance/covariance.cpp:186]:
```suggestion
                   .def("get_cpu_macro_block", &compute_hyperparams_t::get_cpu_macro_block);
#if defined(ONEDAL_VERSION) && ONEDAL_VERSION >= 20250700
                        })
                   .def("set_cpu_grain_size", &compute_hyperparams_t::set_cpu_grain_size)
                   .def("get_cpu_grain_size", &compute_hyperparams_t::get_cpu_grain_size);
```

**Review by icfaust — COMMENTED** [review]
Looks relatively ready to go, code is easy to read and follows convention. Rather than using lambdas, you can reference the hyperparameter type's methods directly.

File: onedal/covariance/covariance.cpp [hunk 4]
```diff
@@ -173,8 +173,18 @@ void init_compute_hyperparameters(py::module_& m) {
...
+#endif
                    });
```
Comment (icfaust) [inline onedal/covariance/covariance.cpp]:
```suggestion
                   .def("get_cpu_macro_block",&compute_hyperparams_t::get_cpu_macro_block);
#if defined(ONEDAL_VERSION) && ONEDAL_VERSION >= 20250700
                        })
                   cls.def("set_cpu_grain_size",&compute_hyperparams_t::set_cpu_grain_size)
                   .def("get_cpu_grain_size", &compute_hyperparams_t::get_cpu_grain_size);
#endif
```

File: generator/gen_daal4py.py [hunk 5]
```diff
@@ -1223,14 +1223,21 @@ def gen_daal4py(dalroot, outdir, version, warn_all=False, no_dist=False, no_stre
...
         for filename in filenames:
-            call(
-                [
-                    shutil.which("clang-format"),
-                    "-i",
-                    jp(dirpath, filename),
-                    "-style=file:" + formatfile,
-                ]
-            )
+            try:
```
Comment (icfaust) [inline generator/gen_daal4py.py:1226]: Thank you for doing this. Its one of those hidden things for development that catches me from time to time.  Nitpick would be to add a note in the PR description

**Review by icfaust — COMMENTED** [review]
I reran the timed-out windows CI job, and make sure to finish the checklist. I think with a private CI run and @david-cortes-intel 's point addressed we can approve this.

File: generator/gen_daal4py.py (hunk 5 above)
Comment (Vika-F) [reply generator/gen_daal4py.py:1226]: Yes, I also have problems with it from time to time. Decided to put an end to those :)
I've updated the description.

File: sklearnex/decomposition/tests/test_pca.py (hunk 1 above)
Comment (Vika-F) [reply sklearnex/decomposition/tests/test_pca.py]: @david-cortes-intel I've made the modifications, but I am not sure about their correctness. Can you please take a look?
There is also some copy-paste which I hope will be removed after fixing the hyperparameters logic.
